### PR TITLE
Dissolution ux fixes/12629

### PR DESCRIPTION
**Pull request by meganawong**
*Issue #:* /bcgov/entity#12629

*Description of changes:*
Implement UX fixes and update validation errors for dissolution dashboard.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (severinbeauvais) [inline src/views/DissolutionFirm/DissolutionFirm.vue:196]: Always invalid?

File: src/components/common/EntityInfo.vue
Comment (severinbeauvais) [inline src/components/common/EntityInfo.vue:112]: What is the requirement here?

Comment (meganawong) [reply src/components/common/EntityInfo.vue:112]: I was just advised to remove the DBA title from the header in the Dissolution Dashboard for SP

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue]: Fixed

File: src/store/getters/state-getters.ts
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts:472]: Unfortunately lint doesn't catch this, but please indent continuing lines.

By the way, an easy-to-read format for ternaries is like this:
```
const isEffectiveDateTimeValid = (isBaseCompany(state) || isTypeFirm(state))
  ? getEffectiveDateTime(state).valid
  : true
```

**Comment by severinbeauvais** [comment]
I will wait until you mark this PR ready for review before I approve it.

File: src/components/common/EntityInfo.vue
Comment (severinbeauvais) [reply src/components/common/EntityInfo.vue:112]: For dissolutions only? This header is also used for firm registrations.

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (severinbeauvais) [inline src/views/DissolutionFirm/DissolutionFirm.vue:460]: Can you use the base class `invalid-section` here instead of duplicating things?

Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: the invalid-section class includes turning the title-labels red, which i do not need in this case

Comment (severinbeauvais) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: Are you doing anything with class "title-label"? If not, you could just remove it.

Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: oops thanks, will remove

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (severinbeauvais) [inline src/views/DissolutionFirm/DissolutionFirm.vue:321]: So, dissolution date is always invalid?

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (severinbeauvais) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: Can you remove this CSS now?

Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: the standard invalid-section class turns the labels red in addition to adding the red side bar but there are certain cases  where I do not need both

Comment (severinbeauvais) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: I thought you got around that by removing the `title-label` class from your label?

Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: the special class is used in the documents delivery section as well

Comment (severinbeauvais) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: OK, I see.

I would prefer to see an override for title-label in DocumentDelivery for this on-off special case instead of duplicating the base CSS here. (You can do this using `::v-deep`.)

Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: I see, I can only find examples of how to alter or add additional styling within v-deep online. How can I remove the title-label styling in the override entirely?

Comment (severinbeauvais) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: There are 62 examples of using v-deep in Create UI.

In short, you need to "reach into" (using v-deep) the DocumentDelivery component and override the colour CSS property.

Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: thanks will switch it

Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: Screen Shot 2022-07-11 at 2 06 12 PM

Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: Screen Shot 2022-07-11 at 2 21 45 PM

Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: Screen Shot 2022-07-11 at 2 23 03 PM

Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: trying to override the title-label styling in the invalid-section within the document-delivery-section as you suggested, is my approach wrong

Comment (severinbeauvais) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: I see 3 places that the code sets the "invalid-section" class on the DocumentDelivery component, so you may need this override in all places.

I think this CSS might work:
```
// override the title labels so they don't change colour on error
::v-deep #document-delivery .title-label {
  color: $gray7 !important;
}
```

Or maybe you will need to more precisely select the label (since CSS has an order of operation).

Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:460]: // override the title labels so they don't change colour on error
::v-deep #document-delivery-section .invalid-section.title-label {
  color: $gray7 !important;
}

i tried what you suggested as well as the snippet of code above with more specific labels, neither appear to work, unsure if i am missing something else

File: src/views/DissolutionFirm/DissolutionFirm.vue
Comment (meganawong) [reply src/views/DissolutionFirm/DissolutionFirm.vue:537]: oops forgot to paste over, done

**Review by severinbeauvais — APPROVED** [review]

**Review by shabeeb-aot — APPROVED** [review]

**Pull request merged**
